docs/src/developing/clients/jsonrpc-api.md

```diff
@@ -389,6 +389,7 @@ Returns identity and transaction information about a confirmed block in the ledg
   - (optional) `transactionDetails: <string>` - level of transaction detail to return, either "full", "signatures", or "none". If parameter not provided, the default detail level is "full".
   - (optional) `rewards: bool` - whether to populate the `rewards` array. If parameter not provided, the default includes rewards.
   - (optional) [Commitment](jsonrpc-api.md#configuring-state-commitment); "processed" is not supported. If parameter not provided, the default is "finalized".
+  - (optional) `maxSupportedTransactionVersion: <number>` - set the max transaction version to return in responses. If the requested block contains a transaction with a higher version, an error will be returned.
```

Comment:
Sorry, I might be being dense, but I'm not sure I follow why this is necessary for either `getBlock` and `getTransaction`? We generally consider adding fields to rpc responses as non-breaking, and it looks to me like all the new rpc fields are additive (`loadedAddresses` and `version` for blocks, `addressTableLookups` for transactions).

Reply:
Is it that you think it's important for clients to receive and error rather than a v0 block/transaction that is just missing those fields? If so, I don't think I agree.

Reply:
My intent here is that we can totally diverge from keeping backwards compatible fields for future transaction versions if we want.

Reply:
That said, I will start giving this a proper review now :)

Reply:
What do you envision the client upgrade / feature-activation process looking like? Basically every exchange will be affected.